packages/editor/src/components/post-actions/reset-posts-page.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useMemo } from '@wordpress/element';
+import {
+	Button,
+	__experimentalText as Text,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import { store as noticesStore } from '@wordpress/notices';
+
+/**
+ * Internal dependencies
+ */
+import { getItemTitle } from '../../utils';
+
+const ResetPostsPageModal = ( { items, closeModal } ) => {
+	const [ item ] = items;
+	const pageTitle = getItemTitle( item );
+	const { isPageOnFrontSet, isSaving } = useSelect( ( select ) => {
+		const { getEntityRecord, isSavingEntityRecord } = select( coreStore );
+		const siteSettings = getEntityRecord( 'root', 'site' );
+		return {
+			isPageOnFrontSet: siteSettings?.page_on_front !== 0,
+			isSaving: isSavingEntityRecord( 'root', 'site' ),
+		};
+	} );
+
+	const { saveEditedEntityRecord, saveEntityRecord } =
+		useDispatch( coreStore );
+	const { createSuccessNotice, createErrorNotice } =
+		useDispatch( noticesStore );
+
+	async function onResetPostsPage( event ) {
+		event.preventDefault();
+
+		try {
+			// Save new posts page settings.
+			await saveEditedEntityRecord( 'root', 'site', undefined, {
+				page_for_posts: 0,
+				show_on_front: isPageOnFrontSet ? 'page' : 'posts',
+			} );
+
+			// This second call to a save function is a workaround for a bug in
+			// `saveEditedEntityRecord`. This forces the root site settings to be updated.
+			// See https://github.com/WordPress/gutenberg/issues/67161.
+			await saveEntityRecord( 'root', 'site', {
+				page_for_posts: 0,
+				show_on_front: isPageOnFrontSet ? 'page' : 'posts',
+			} );
+
+			createSuccessNotice( __( 'Posts page reset' ), {
+				type: 'snackbar',
+			} );
+		} catch ( error ) {
```

Comment:
Is this exception error actually possible? Because saveEntityRecord doesn't seem to throw any exceptions by default.

https://github.com/WordPress/gutenberg/blob/add/more-homepage-actions/packages/core-data/README.md#saveentityrecord

Reply:
I think it's nice to have the `catch` block anyway, to catch any other errors (like a network error maybe). But I completely see your point, I don't think `saveEntityRecord` throws any exceptions.